xarray/core/dataarray.py

```diff
@@ -761,11 +761,15 @@ def data(self, value: Any) -> None:
     @property
     def values(self) -> np.ndarray:
         """
-        The array's data as a numpy.ndarray.
+        The array's data converted to numpy.ndarray.
 
-        If the array's data is not a numpy.ndarray this will attempt to convert
-        it naively using np.array(), which will raise an error if the array
-        type does not support coercion like this (e.g. cupy).
+        Note that this array is not copied; operations on it follow
+        numpy's rules of what generates a view vs. a copy, and changes
+        to this array may be reflected in the DataArray as well.
```

Comment:
To make this easier to follow, should we swap the order of the two paragraphs? That way, we first state that we convert (which might fail for certain wrapped types), and then point out that the result is not a copy.

Reply:
Sounds good, implemented in the newest (https://github.com/pydata/xarray/pull/8744/commits/327228b3ca99b248ebcf2a75a67ac2a3882ae14d)